pkg/mqtt5/client.go

```diff
@@ -61,69 +49,82 @@ func NewMqtt5Client(config config.Mqtt5Config) Mqtt5Client {
-		c.configuration.AuthMode,
-		c.configuration.SecretName)
-	if secretProvider == nil {
-		logger.Error("No secret provider")
+	authMode := strings.ToLower(c.configuration.AuthMode)
+	if len(authMode) == 0 || authMode == secret.AuthModeNone {
 		return nil
 	}
-	secrets, err := secretProvider.GetSecret(c.configuration.SecretName)
+
+	if len(c.configuration.SecretName) == 0 {
+		return errors.New("missing SecretName")
+	}
+
+	logger.Infof("Setting auth data for secure MessageBus with AuthMode='%s' and SecretName='%s",
+		authMode,
+		c.configuration.SecretName)
+
+	secretData, err := secret.GetSecretData(c.configuration.SecretName, secretProvider)
 	if err != nil {
 		return fmt.Errorf("Unable to get Secret Data for secure message bus: %w", err)
 	}
 
-	switch c.configuration.AuthMode {
-	case AuthModeUsernamePassword:
-		if secrets[SecretUsernameKey] == "" || secrets[SecretPasswordKey] == "" {
+	switch authMode {
+	case secret.AuthModeUsernamePassword:
+		if secretData.Username == "" || secretData.Password == "" {
 			return fmt.Errorf("AuthModeUsernamePassword selected however Username or Password was not found for secret=%s", c.configuration.SecretName)
 		}
-		c.authData.Username = secrets[SecretUsernameKey]
-		c.authData.Password = secrets[SecretPasswordKey]
+		c.authData.Username = secretData.Username
+		c.authData.Password = secretData.Password
 		c.connect.Username = c.authData.Username
 		c.connect.Password = []byte(c.authData.Password)
 		c.connect.UsernameFlag = true
 		c.connect.PasswordFlag = true
-	case AuthModeCert:
-		if secrets[SecretClientCert] == "" || secrets[SecretClientKey] == "" {
+	case secret.AuthModeCert:
+		if secretData.KeyPemBlock == "" || secretData.CertPemBlock == "" {
 			return fmt.Errorf("AuthModeCert selected however the key or cert PEM block was not found for secret=%s", c.configuration.SecretName)
 		}
-		c.authData.CertPemBlock = secrets[SecretClientCert]
-		c.authData.KeyPemBlock = secrets[SecretClientKey]
-	case AuthModeCA:
-		if secrets[SecretCACert] == "" {
+		c.authData.CertPemBlock = secretData.CertPemBlock
+		c.authData.KeyPemBlock = secretData.KeyPemBlock
+	case secret.AuthModeCA:
+		if secretData.CaPemBlock == "" {
 			return fmt.Errorf("AuthModeCA selected however no PEM Block was found for secret=%s", c.configuration.SecretName)
 		}
 
-	case AuthModeNone:
-		// Nothing to validate
 	default:
 		return fmt.Errorf("Invalid AuthMode of '%s' selected", c.configuration.AuthMode)
 	}
 
-	if len(secrets[SecretCACert]) > 0 {
+	if len(secretData.CaPemBlock) > 0 {
 		caCertPool := x509.NewCertPool()
-		ok := caCertPool.AppendCertsFromPEM([]byte(secrets[SecretCACert]))
+		ok := caCertPool.AppendCertsFromPEM([]byte(secretData.CaPemBlock))
 		if !ok {
 			return errors.New("Error parsing CA Certificate")
 		}
-		c.authData.CaPemBlock = secrets[SecretCACert]
+		c.authData.CaPemBlock = secretData.CaPemBlock
 	}
 
 	return nil
 }
 
+func (c *Mqtt5Client) SetMutex(mutex *sync.RWMutex) {
+	c.mutex = mutex
+}
```

Comment:
why do we need this function? the mutex should be automatically created in `NewMqtt5Client`

Reply:
updated